src/frontend/js/hooks/useIsLoading.ts

```diff
@@ -0,0 +1,43 @@
+import { useEffect, useState } from 'react';
+
+// await new Promise(resolve => setTimeout(resolve, 1000))
+
+const useIsLoading = (loadingStates: boolean[], delayInMillisecond = 500) => {
+  const [isLoading, setIsLoading] = useState<boolean>(!loadingStates.every((v) => v === false));
+  const [timerId, setTimerId] = useState<ReturnType<typeof setTimeout>>();
```

Comment:
I feel we could use a `useRef` in this case. We don't care to trigger a render when timer id change.

Reply:
moved into #1971